src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -257,12 +300,21 @@
      * This regexp is used to extract the content of a summary javadoc tag.
      */
     private static final Pattern SUMMARY_PATTERN = Pattern.compile("\\{@summary ([\\S\\s]+)}");
+
+    /**
+     * This regexp is used to extract the content of an inline return javadoc tag.
+     */
+    private static final Pattern INLINE_RETURN_PATTERN = Pattern.compile("\\{@return ([\\S\\s]+)}");
```

Comment:
Since this is an `AbstractJavadocCheck`, can you explain why we need to regexp and not use the Javadoc tokens? I am not familar with the history of the other regexp here.

Reply:
I based this off the already-existing `@summary` check, so it uses the same regex and also supports multiple lines like `@summary`. I will validate this in tests to be sure.

I don't know why `@summary` uses this regex instead of the parsing. My suspicion is that it's hard to decode the original text from the AST, e.g. when I tried it with `{@return a {@code 0}}` it gave me a very deeply nested AST that I would need to use some sort of visitor API to fully evaluate for its visible text. I'm not confident I'd use the visitor API right either since I'm not familiar with all the token types and what counts as "visible text".

If you know of a way to retrieve all of the visible text inside a Javadoc AST node, I can change to that instead.